### PR TITLE
Consistent screen-space coordinates

### PR DESCRIPTION
**Pull request by devil-ira**
# Objective

Make the coordinate systems of screen-space items (cursor position, UI, viewports, etc.) consistent.

## Solution

Remove the weird double inversion of the cursor position's Y origin. Once in bevy_winit to the bottom and then again in bevy_ui back to the top.
This leaves the origin at the top left like it is in every other popular app framework.

Update the `world_to_viewport`, `viewport_to_world`, and `viewport_to_world_2d` methods to flip the Y origin (as they should since the viewport coordinates were always relative to the top left).

## Migration Guide

`Window::cursor_position` now returns the position of the cursor relative to the top left instead of the bottom left.
This now matches other screen-space coordinates like `RelativeCursorPosition`, UI, and viewports.

The `world_to_viewport`, `viewport_to_world`, and `viewport_to_world_2d` methods on `Camera` now return/take the viewport position relative to the top left instead of the bottom left.

If you were using `world_to_viewport` to position a UI node the returned `y` value should now be passed into the `top` field on `Style` instead of the `bottom` field.
Note that this might shift the position of the UI node as it is now anchored at the top.

If you were passing `Window::cursor_position` to `viewport_to_world` or `viewport_to_world_2d` no change is necessary.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

File: examples/3d/blend_modes.rs [hunk 1]
```diff
@@ -220,51 +220,34 @@ fn setup(
...
+                    },
+                    ..default()
+                },
+                ExampleLabel { entity },
+            ))
+            .with_children(|parent| {
+                parent.spawn(
+                    TextBundle::from_section(label, label_text_style.clone()).with_style(Style {
+                        position_type: PositionType::Absolute,
+                        bottom: Val::Px(0.),
```
Comment (devil-ira) [inline examples/3d/blend_modes.rs:239]: I had to rework the labels in the blend_modes example a bit to keep their appearance same.
Not a usability regression, it's just that for the labels to display correctly they need to be 'anchored' to the bottom left which happened to align with what `world_to_viewport` returned before.

Solved by adding a parent node and using `bottom: Val::Px(0.),` for the text itself.

Comment (alice-i-cecile) [reply examples/3d/blend_modes.rs:239]: Yeah I'm happy with the changes here!

**Review by superdump — COMMENTED** [review]
I think this looks good but need to recheck when I’m less tired.

**Review by cart — APPROVED** [review]

**Comment by nablabla** [comment]
please provide a formula to convert, guessing is so frustrating

**Pull request merged**